doc/sphinx/language/core/definitions.rst

```diff
@@ -92,30 +104,31 @@ Section :ref:`typing-rules`.
       | {* @binder } : @type
       reduce ::= Eval @red_expr in
 
-   These commands bind :n:`@term` to the name :n:`@ident` in the global environment,
-   provided that :n:`@term` is well-typed.  They can take the :attr:`local` :term:`attribute`,
-   which makes the defined :n:`@ident` accessible only through their fully
-   qualified names, even if :cmd:`Import` or its variants has been used on the
-   current :cmd:`Module`.
+   In its most basic form :g:`Definition @ident := term`, the command
+   binds :n:`@term` to the name :n:`@ident` in the global environment,
+   provided that :n:`@term` is well-typed.
+
+   The form :n:`Definition @ident : @type := @term` checks that the type of :n:`@term`
+   is definitionally equal to :n:`@type`, and registers :n:`@ident` as being of type
+   :n:`@type`, and bound to value :n:`@term`.
+
+   The form :n:`Definition @ident {* @binder } : @type := @term` is equivalent to
+   :n:`Definition @ident : forall {* @binder }, @type := fun {* @binder } => @term`
+   (and similarly when one of `@term` or `@type` is omitted).
+
```

Comment:
```suggestion
   The command
   binds :n:`@term` to the name :n:`@ident` in the global environment,
   provided that :n:`@term` is well-typed.

   If :n:`@type` is specified, the command checks that the type of :n:`@term`
   is definitionally equal to :n:`@type`.

   If @binder is specified, the definition is equivalent to
   :n:`Definition @ident : forall {* @binder }, @type := fun {* @binder } => @term`
   (and similarly when one of `@term` or `@type` is omitted).

```
We prefer describing the behavior of each bit of the syntax separately rather than listing multiple similar forms.  It's easier for readers, who don't have to read a lot of very similar text and mentally do a diff to figure out what each bit of the syntax does.

"and registers :n:`@ident` as being of type :n:`@type`, and bound to value :n:`@term`" -- what is this registration?  Seems like implementation detail that the user probably doesn't need to know about.